### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend operator ReduceMean-11, 13, 18 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20556]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceMean for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMean-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMean-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMean-18)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

First of all, please, take a look on [ReduceMax](https://github.com/openvinotoolkit/openvino/pull/23475) PR for a reference.

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceMean)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMean-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMean-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMean-18)
1. Operator already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by rghvsh** [comment]
.take

**Comment by rghvsh** [comment]
Hey! @gkrivor just wanted to make sure that I got this right:

I need to change the properties of ReduceMean such that they are same in each opset. For example ReduceMean-18 has an extra attribute `noop_with_empty_axes` and 2 inputs instead of 1 unlike in other opsets.So I need to add this to other or remove from this?

Could you also point me to the original framework tried searching but there were too many with same name didn't know which to pick.

Thanks
Raghav

**Comment by p-wysocki** [comment]
Hello @rghvsh, sorry for the late reply, the holiday season just ended. :) @gkrivor could you please take a look?

Also, I am happy to announce that we have created a channel dedicated to Good First Issues support on our [Intel DevHub Discord server](https://discord.gg/wPuqAujS)! Join it to receive support, engage in discussions, ask questions and talk to OpenVINO developers.

**Comment by rghvsh** [comment]
Hey! @gkrivor @p-wysocki  which of the these should I consider as the original framework?

**Comment by mlukasze** [comment]
https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceMean

**Comment by gkrivor** [comment]
@rghvsh hope you are doing great!

Yeah, you a right, if documentation says we have a difference in inputs (input changed to an attribute, or opposite way) - need to implement correct translation function which will do it. But keeping old implementation too, because models which implements older nodes (for example, in a range 13-18 they will use implementation for opset 13) should be also functional.

I've added a link to an issue with similar discussion: https://github.com/openvinotoolkit/openvino/issues/20553

I suggest to identify difference and define it as a table here (would be easier to review later).

You should get something like https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp which will implement a found differences.

As you remember, you will need to add a tests for exact implementations like here: https://github.com/openvinotoolkit/openvino/pull/21825

I mean a prototxt file and corresponing unit test to verify a correctness.

**Comment by p-wysocki** [comment]
Hello @rghvsh, are you still working on that issue?

**Comment by rghvsh** [comment]
Hey!  @p-wysocki yes

**Comment by rghvsh** [comment]
![Image](https://github.com/openvinotoolkit/openvino/assets/116428320/571ea979-1ef7-4a63-980c-4485ac54baac)

**Comment by gkrivor** [comment]
@rghvsh great! as you can see most important change is in opset-18. opset-11 and -13 has a different set of supported input types.
Now you can re-use existing logic to prepare a PR with changes. Be notified some tests may unexpectedly fail on CPU due to broken shape inference, I'll fix it soon.

**Comment by rghvsh** [comment]
PR #23148
